sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -305,34 +344,42 @@ export class EventHubConsumerClient {
 
   /**
    * Provides the Event Hub runtime information.
-   * @param [options] The set of options to apply to the operation call.
-   * @returns A promise that resolves with EventHubProperties.
-   * @throws {Error} Thrown if the underlying connection has been closed, create a new EventHubClient.
-   * @throws {AbortError} Thrown if the operation is cancelled via the abortSignal.
+   * @param options The set of options to apply to the operation call.
+   * @returns A promise that resolves with information about the Event Hub instance.
+   * @throws Error if the underlying connection has been closed, create a new EventHubConsumerClient.
+   * @throws AbortError if the operation is cancelled via the abortSignal.
    */
   getEventHubProperties(options: GetEventHubPropertiesOptions = {}): Promise<EventHubProperties> {
     return this._eventHubClient.getProperties(options);
   }
 
   /**
-   * Subscribe to all messages from all available partitions.
+   * Subscribe to events from all partitions.
+   * 
+   * If checkpoint store is provided to the `EventHubConsumerClient` and there are multiple
+   * instances of your application, then each instance will subscribe to a subset of the
+   * partitions such that the load is balanced amongst them.
    *
-   * @param handlers Handlers for the lifecycle of the subscription - initialization
+   * @param handlers Handlers for the lifecycle of the subscription - subscription initialization
    *                 per partition, receiving events, handling errors and the closing
    *                 of a subscription per partition.
-   * @param options Options to handle additional events related to partitions (errors,
-   *                opening, closing) as well as batch sizing.
+   * @param options A set of options to configure the way events are received.
+   * Most common are `maxBatchSize` and `maxWaitTimeInSeconds` that control the flow of 
+   * events to the handler provided to receive events. A simple usage can be
+   * `{ maxBatchSize: 20, maxWaitTimeInSeconds: 120 }
    */
   subscribe(handlers: SubscriptionEventHandlers, options?: SubscribeOptions): Subscription; // #1
```

Comment:
We can call out the `TypeError` for an invalid `startPosition` now that #6729 is merged!

Reply:
@bterlson had this suggestion of calling out errors that one would want to catch and react programatically to. TypeError would mostly require user to rewrite their code anyway.